py-polars/polars/functions/as_datatype.py

```diff
@@ -113,6 +112,36 @@ def datetime_(
     │ 2     ┆ 5   ┆ 13   ┆ 30     ┆ 2024-02-05 13:30:00 AEDT       │
     │ 3     ┆ 6   ┆ 14   ┆ 45     ┆ 2024-03-06 14:45:00 AEDT       │
     └───────┴─────┴──────┴────────┴────────────────────────────────┘
+
+    We can also use `pl.datetime` for filtering. Notice that `pl.datetime` allows you
+    to overcome The Python Standard Library's limitation of not supporting nanoseconds
+    for `datetime.datetime`.
+
+    >>> df = pl.DataFrame(
+    ...     {
+    ...         'start': [
+    ...             '2024-01-01T01:04:07.123456789',
+    ...             '2024-01-01T02:05:08.234567890',
+    ...             '2024-01-01T03:06:09.345678901',
+    ...         ],
+    ...         "end": [
+    ...             '2024-05-01T04:07:00.456789012',
+    ...             '2024-06-01T05:08:01.567890123',
+    ...             '2024-07-01T06:09:02.678901234',
+    ...         ],
+    ...     })
+    >>> df = df.with_columns(pl.col('start').str.to_datetime(time_unit='ns'))
+    >>> df = df.with_columns(pl.col('end').str.to_datetime(time_unit='ns'))
+    >>> df.filter(pl.col("end") > pl.datetime(2024, 6, 1))
```

Comment:
shall we include `nanoseconds=` in `pl.datetime` here, so it's easier to notice?

Reply:
I added `time_unit="ns"` because there is no parameter `nanoseconds`. Maybe it's possible to add this parameter? (I would have to have an in depth look at the code to check this.)

Reply:
Also, let me know if line 133 and 134 can be optimised because they are currently a bit repetitive. I included two columns to make it clear that this is a dataframe and not a series. However this might be clear enough via line 120 so maybe one column is enough for this example.

Reply:
>  there is no parameter nanoseconds

I'm so sorry 🤦  I'd misremembered, I should've checked

Let's remove
```
Notice that `pl.datetime` allows you
    to overcome The Python Standard Library's limitation of not supporting nanoseconds
    for `datetime.datetime`.
```
and this new change, and then this should be good to go? 🚀